aten/src/ATen/native/cuda/ComplexKernel.cu

```diff
@@ -0,0 +1,49 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/TensorFactories.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cuda/Loops.cuh>
+
+// NOTE: CUDA on Windows requires that the enclosing function
+// of a __device__ lambda not have internal linkage.
+
+namespace at {
+namespace native {
+namespace {
+
+void complex_kernel_cuda(TensorIterator& iter) {
+  // Necessary since comparison_op doesn't actually promote types for CUDA
+  // tensors
+  AT_DISPATCH_FLOATING_TYPES(
+      promote_types(iter.input_dtype(0), iter.input_dtype(1)),
```

Comment:
You don't want to promote the types here, right? In the earlier functions you're already guaranteeing these have the same dtype, either float32 or float64.

Reply:
Yep, thanks.